app/policies/rent_policy.rb

```diff
@@ -0,0 +1,17 @@
+class RentPolicy < ApplicationPolicy
+  def index?
+    user.id == record.first.user_id
```

Comment:
Shouldn't all records have the same user_id?
And also, what you are trying to compare is the user_id you receive from the params vs the one who is logged in. Wouldn't it be better to compare that instead of the user in the models you got from the query? The query should be tested and only bring models from the user who's param was received.

Reply:
**user.id** is the ID of the current_user
**record.first.user_id** is the ID of the user that comes in the response from the endpoint, for example: if y use this endpoint `/api/v1/user/2/rent` I will only receive the rents from user 2


If I login with the user with id 2 and make a query to this endpoint `/api/v1/user/2/rent` there would be no problem, I would receive the answers from the rents

but if I login with a user with another id other than 2, I would receive a response saying that I do not have permissions to see the data, for this example I would receive the next response: `{ error: 'You are not authorized to perform this action.' }`